addons/analytic/models/analytic_account.py

```diff
@@ -125,6 +125,15 @@ def _compute_display_name(self):
                 name = f'{name} - {analytic.partner_id.commercial_partner_id.name}'
             analytic.display_name = name
 
+    @api.constrains('name')
+    def _check_unique_name_in_plan(self):
+        for account in self:
+            account_ids = account.plan_id.account_ids.ids
+            account_ids.remove(account.id)
+            analytic_accounts = self.env['account.analytic.account'].browse(account_ids)
+            if account.name in [a['name'] for a in analytic_accounts]:
+                raise UserError("Another entry with the same name already exists in the same plan.")
```

Comment:
```suggestion
                raise UserError(_("Another entry with the same name already exists in the same plan."))
```

Reply:
hi @aboo-odoo seems you missed to make it translatable

Reply:
Fixed, thanks!

Reply:
is it better to simplify the constrains as follows:
duplicate_accounts = self.env['account.analytic.account'].search([
            ('plan_id', '=', account.plan_id.id),
            ('name', '=', account.name),
            ('id', '!=', account.id)
        ], limit=1)
        if duplicate_accounts: